sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -10,58 +10,38 @@
 
 namespace Azure.AI.TextAnalytics
 {
-    internal static class Transforms
+    internal static partial class Transforms
     {
-        #region Common
+        #region Needs Review
 
-        public static readonly Regex _targetRegex = new Regex("#/tasks/(keyPhraseExtractionTasks|entityRecognitionPiiTasks|entityRecognitionTasks|entityLinkingTasks|sentimentAnalysisTasks|extractiveSummarizationTasks|customSingleClassificationTasks|customMultiClassificationTasks|customEntityRecognitionTasks)/(\\d+)", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
+        //
+        // REGION: Common
+        //
+        internal static TextAnalyticsError ConvertToError(TextAnalyticsErrorInternal error) => throw new NotImplementedException("I think the TextAnalyticsInternalType is no longer used.");
+        internal static List<TextAnalyticsError> ConvertToErrors(IReadOnlyList<TextAnalyticsErrorInternal> internalErrors) => throw new NotImplementedException("I think the TextAnalyticsInternalType is no longer used.");
 
-        internal static TextAnalyticsError ConvertToError(TextAnalyticsErrorInternal error)
-        {
-            string errorCode = error.Code;
-            string message = error.Message;
-            string target = error.Target;
-            InnerError innerError = error.Innererror;
+        #endregion
 
-            if (innerError != null)
-            {
-                // Return the innermost error, which should be only one level down.
-                return new TextAnalyticsError(innerError.Code, message, target);
-            }
+        #region Common
 
-            return new TextAnalyticsError(errorCode, message, target);
-        }
+        public static readonly Regex _targetRegex = new Regex("#/tasks/(keyPhraseExtractionTasks|entityRecognitionPiiTasks|entityRecognitionTasks|entityLinkingTasks|sentimentAnalysisTasks|extractiveSummarizationTasks|customSingleClassificationTasks|customMultiClassificationTasks|customEntityRecognitionTasks)/(\\d+)", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
 
-        internal static TextAnalyticsError ConvertToError(Error error)
+        internal static TextAnalyticsError ConvertToError(DocumentError error)
```

Comment:
from looking at the PR is hard to see what the difference between `ConvertToError(DocumentError error)` and ConvertToError(Error error)`` is. Do you happen to know?

Reply:
Not sure that I understand.  They're different types with no common ancestor, so they needed to be overloaded unless we're going to unify the types.

Reply:
I believe I understand what is happening here. `DocumentError` contains `Error`. It also has an `Id` property. in our client logic, we always look at the `Id` of the document first and then assign to that `Document` the appropriate `Error`.
so if this is true, no one should call `ConvertToError(DocumentError error)` .

Is hard to see if this is true from the PR. Can you see if someone is actually calling that method?

Reply:
This looks to be a hold-over from before Nisha's latest merge that I reformatted as I was adding the overloads.   _([src](https://github.com/Azure/azure-sdk-for-net/blob/08e2e0401ece971c0d560991f38c4373b6ad7629/sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs#L36))_    It does not appear that there are any references to it.   I'll remove it for now; it's simple enough to add back in if we end up needing it somewhere in the future.